source/h2_stream.c

```diff
@@ -747,6 +747,10 @@ int aws_h2_stream_on_activated(struct aws_h2_stream *stream, enum aws_h2_stream_
     stream->thread_data.window_size_self =
         connection->thread_data.settings_self[AWS_HTTP2_SETTINGS_INITIAL_WINDOW_SIZE];
 
+    if (!connection->base.stream_manual_window_management) {
+        stream->thread_data.window_size_self_dropped_threshold =
+            connection->thread_data.settings_self[AWS_HTTP2_SETTINGS_INITIAL_WINDOW_SIZE] / 2;
```

Comment:
similar feedback to connection window batching:
- we should probably still batch up WINDOW_UPDATE frames, even if the user is doing it "manual", because most users will do 1 update for each read(), just like we were doing in our "automatic" behavior, which was giving performance problems
- we probably want to send the updates anyway if the total window size is too small

maybe just hardcode these 2 magic numbers: the preferred batch size, and the window size at which we don't bother batching, and use the same values for both stream and connection behavior, whether its on automatic or manual mode